identity-apps-core/react-ui-core/src/components/adapters/checkbox-field-adapter.js

```diff
@@ -17,42 +17,60 @@
  */
 
 import PropTypes from "prop-types";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Checkbox } from "semantic-ui-react";
+import useFieldValidation from "../../hooks/use-field-validations";
 import { useTranslations } from "../../hooks/use-translations";
 import { resolveElementText } from "../../utils/i18n-utils";
 import Hint from "../hint";
+import ValidationError from "../validation-error";
 
-const CheckboxFieldAdapter = ({ component, formStateHandler }) => {
+const CheckboxFieldAdapter = ({ component, formState, formStateHandler, fieldErrorHandler }) => {
 
-    const { identifier, label, hint } = component.config;
+    const { identifier, label, hint, validations, required } = component.config;
     const { translations } = useTranslations();
+    const [ value, setValue ] = useState(false);
+    const { fieldErrors, validate } = useFieldValidation(validations, value);
+
+    const handleFieldValidation = (value) => {
+        const { errors, isValid } = validate({ identifier, required }, value);
+
+        fieldErrorHandler(identifier, isValid ? null : errors);
+    };
+
+    const handleChange = (e, data) => {
+        setValue(data.checked);
+        handleFieldValidation(data.checked);
+    };
```

Comment:
```suggestion
    const handleChange = (e, data) => {
        setValue(data.checked);
        handleFieldValidation(data.checked);
    };
```
shall we remove thie value state and directly add the on change handler